File: 2743-count-substrings-without-repeating-character/2743-count-substrings-without-repeating-character.cs
public class Solution {
    public int NumberOfSpecialSubstrings(string s) {
        Dictionary<char, int> test = new Dictionary<char, int> ();
        int res = 0;
        int left = 0;
        for (int right = 0; right < s.Length; right++) {
            if (test.ContainsKey(s[right])) {
                left = Math.Max(left, test[s[right]]+1);
            }
            res += right - left + 1;
            test[s[right]] = right;
        }
        return res;
    }
}

File: 2743-count-substrings-without-repeating-character/2743-count-substrings-without-repeating-character.py
class Solution:
    def numberOfSpecialSubstrings(self, s: str) -> int:
        seen = {}
        left = 0
        res = 0
        for right in range(len(s)):
            if s[right] in seen:
                left = max(seen[s[right]]+1, left)
            seen[s[right]] = right
            
            res += right - left + 1
        return res
        